homeassistant/components/overkiz/water_heater_entities/atlantic_domestic_hot_water_production_mlb_component.py

```diff
@@ -116,46 +118,35 @@ def current_operation(self) -> str:
             cast(str, self.executor.select_state(OverkizState.MODBUSLINK_DHW_MODE))
             == OverkizCommandParam.MANUAL_ECO_INACTIVE
         ):
-            return OverkizCommandParam.MANUAL
+            # STATE_ELECTRIC is a substitution for OverkizCommandParam.MANUAL
+            # to keep up with the conventional state usage only
+            # https://developers.home-assistant.io/docs/core/entity/water-heater/#states
+            return STATE_ELECTRIC
 
         return STATE_OFF
 
     async def async_set_operation_mode(self, operation_mode: str) -> None:
         """Set new operation mode."""
-        if operation_mode in (STATE_PERFORMANCE, OverkizCommandParam.BOOST):
+        if operation_mode == STATE_PERFORMANCE:
             if self.is_away_mode_on:
                 await self.async_turn_away_mode_off()
             await self.async_turn_boost_mode_on()
-        elif operation_mode in (
-            OverkizCommandParam.ECO,
-            OverkizCommandParam.MANUAL_ECO_ACTIVE,
-        ):
+        elif operation_mode == STATE_ECO:
             if self.is_away_mode_on:
                 await self.async_turn_away_mode_off()
             if self.is_boost_mode_on:
                 await self.async_turn_boost_mode_off()
             await self.executor.async_execute_command(
                 OverkizCommand.SET_DHW_MODE, OverkizCommandParam.AUTO_MODE
             )
-        elif operation_mode in (
-            OverkizCommandParam.MANUAL,
-            OverkizCommandParam.MANUAL_ECO_INACTIVE,
-        ):
+        elif operation_mode == STATE_ELECTRIC:
             if self.is_away_mode_on:
                 await self.async_turn_away_mode_off()
             if self.is_boost_mode_on:
                 await self.async_turn_boost_mode_off()
             await self.executor.async_execute_command(
                 OverkizCommand.SET_DHW_MODE, OverkizCommandParam.MANUAL_ECO_INACTIVE
             )
```

Comment:
I think we need to handle `STATE_OFF` too as operation mode input.

Reply:
not for this device, as there's no such thing as `STATE_OFF` for this device, it's just `away`.
done.

Reply:
I don't understand your reply. It has `STATE_OFF` as part of its operation modes and you did add a case for it now in the service handler here.